FreydCategoriesForCAP/gap/FreydCategory.gi

```diff
@@ -1745,6 +1745,233 @@ InstallGlobalFunction( INSTALL_FUNCTIONS_FOR_FREYD_CATEGORY,
       
     fi;
     
+    ## Random Methods by Lists
+    ##
+    if ForAll( [ "RandomObjectByList",
+                 "RandomMorphismWithFixedSourceByList",
+                 "RandomMorphismWithFixedRangeByList",
+                 "RandomMorphismWithFixedSourceAndRangeByList",
+                 "RandomMorphismByList",
+                 "DirectSumMorphismToWeakBiPushout",
+                 "WeakBiFiberProductMorphismToDirectSum" ], m -> CanCompute( underlying_category, m ) ) then
+      
+      ##
+      AddRandomObjectByList( category,
+        function ( cat, L )
+          
+          return FreydCategoryObject( cat, RandomMorphismByList( underlying_category, L ) );
+          
+      end );
+      
+      ##
+      AddRandomMorphismWithFixedSourceByList( category,
+        function ( cat, S, L )
+          local rho_S, rho, delta, mu;
+          
+          rho_S := RelationMorphism( S );
+          
+          rho := RandomMorphismWithFixedSourceByList( underlying_category, Source( rho_S ), L );
+          
+          delta := DirectSumMorphismToWeakBiPushout( underlying_category, rho_S, rho );
+          
+          delta := PreCompose( delta, RandomMorphismWithFixedSourceByList( underlying_category, Range( delta ), L ) );
+          
+          mu := List( [ 1, 2 ], i -> PreCompose(
+                                        underlying_category,
+                                        InjectionOfCofactorOfDirectSum( underlying_category, [ Range( rho_S ), Range( rho ) ], i ),
+                                        delta ) );
```

Comment:
You can make this even simpler:

`PreCompose( InjectionOfCofactorOfDirectSum, DirectSumMorphismToWeakBiPushout )` should simply be `InjectionOf{First,Second}CofactorOfWeakBiPushout`.

Reply:
Updated!